frontend/src/scenes/settings/types.ts

```diff
@@ -110,6 +110,7 @@ export type Setting = {
      */
     flag?: FeatureFlagKey | `!${FeatureFlagKey}`
     features?: AvailableFeature[]
+    hidden?: 'cloud'
```

Comment:
nit: I don't like the name of this, should be something like `hideOn` and take a list of enum values from https://github.com/PostHog/posthog/blob/8fd394031bcbb12021a350c39d5dc773ed12c660/frontend/src/types.ts#L204-L205

Otherwise the way it is now isn't very reusable.